tests/integration/api/test_user.py

```diff
@@ -272,9 +272,13 @@ def test_delete_user(
 ):
     user_id = user.id
     response = client.delete(f"{api_url_v1}/users/{user_id}/")
-    assert response.status_code == status.HTTP_204_NO_CONTENT
 
-    assert models.User.objects.filter(id=user_id).count() == 0
+    if user_id != user.id:
```

Comment:
I think this kind of conditionals in tests are not a good practice and might result in a lack of coverage and a false sense of confidence.
A test should be, in general, single case scoped. When the logic of the test doesn't change across the cases, when for example you only want to test a set of different inputs, [pytest parametrization](https://docs.pytest.org/en/7.3.x/how-to/parametrize.html) is a good choice. 

But in this case It looks like we should have a different test.

Reply:
Got it! It really makes sense. I will update the test.